homeassistant/components/image_processing/__init__.py

```diff
@@ -0,0 +1,220 @@
+https://home-assistant.io/components/image_processing/
+"""
+import asyncio
+import logging
+import os
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_ENTITY_PICTURE, CONF_TIMEOUT,
+    CONF_NAME, CONF_ENTITY_ID, STATE_UNKNOWN)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.util.async import run_coroutine_threadsafe
+
+DOMAIN = 'image_processing'
+DEPENDENCIES = ['camera']
+
+_LOGGER = logging.getLogger(__name__)
+
+PROCESSING_CLASSES = [
+    None,            # Generic
+    'alpr',          # Processing licences plates
+    'face',          # Processing biometric face data
+]
+
+SERVICE_SCAN = 'scan'
+
+EVENT_FOUND_PLATE = 'found_plate'
+
+ATTR_PLATE = 'plate'
+ATTR_CONFIDENCE = 'confidence'
+
+CONF_SOURCE = 'source'
+CONF_CONFIDENCE = 'confidence'
+
+DEFAULT_TIMEOUT = 10
+DEFAULT_CONFIDENCE = 80
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SOURCE): vol.All(cv.ensure_list, [vol.Schema({
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Optional(CONF_NAME): cv.string,
+    })]),
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+    vol.Optional(CONF_CONFIDENCE, default=DEFAULT_CONFIDENCE):
+        vol.All(vol.Coerce(float), vol.Range(min=0, max=100))
+})
+
+SERVICE_SCAN_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+
+def scan(hass, entity_id=None):
+    """Force process a image."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_SCAN, data)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup image processing."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+
+    yield from component.async_setup(config)
+
+    descriptions = yield from hass.loop.run_in_executor(
+        None, load_yaml_config_file,
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+
+    @asyncio.coroutine
+    def async_scan_service(service):
+        """Service handler for scan."""
+        image_entities = component.async_extract_from_service(service)
+
+        update_task = [entity.async_update_ha_state(True) for
+                       entity in image_entities]
+        if update_task:
+            yield from asyncio.wait(update_task, loop=hass.loop)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_SCAN, async_scan_service,
+        descriptions.get(SERVICE_SCAN), schema=SERVICE_SCAN_SCHEMA)
+
+    return True
+
+
+class ImageProcessingEntity(Entity):
+    """Base entity class for image processing."""
+
+    camera_entity = None  # Load pictre from
```

Comment:
I don't create a init only for set camera_entity. I think we can that move to platform. Or not?